docker/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM gradle:jdk17 AS assemble_plugin
+
+WORKDIR /build
+COPY ./opensearch-remote-processor/ ./opensearch-remote-processor/
+WORKDIR /build/opensearch-remote-processor
+RUN ./gradlew assemble
+
+FROM opensearchproject/opensearch:2.12.0 AS opensearch
+COPY --from=assemble_plugin /build/opensearch-remote-processor/build/distributions/remote-processor-2.12.0-SNAPSHOT.zip /tmp/
+RUN /usr/share/opensearch/bin/opensearch-plugin install --batch file:///tmp/remote-processor-2.12.0-SNAPSHOT.zip
```

Comment:
By versioning I meant being able to figure out the commit that the plugin was built at.  Something similar to the GIT_BRANCH GIT_COMMIT GIT_DIFF stuff done in all the containers. We'll need two versions for the opensearch container now. One for the main container and one for the information for the RPS.

I'm ok with ignoring this here since I expect we're going to change the way we do the build to build it as part of sycamore and have RPS as a sub-module (since we want to have one opensearch container).